strawberryfields/backends/fockbackend/tests/test_homodyne.py

```diff
@@ -31,37 +26,48 @@ class TestRepresentationIndependent:
 
     def test_mode_reset_vacuum(self, setup_backend, tol):
         """Tests that modes get reset to the vacuum after measurement."""
-
         backend = setup_backend(1)
 
-        backend.squeeze(SQUEEZE_VAL, 0)
-        backend.displacement(MAG_ALPHAS[-1], 0)
+        r = np.arcsinh(1.0)
+        alpha = 0.8
+
+        backend.squeeze(r, 0)
+        backend.displacement(alpha, 0)
         backend.measure_homodyne(0, 0)
+
         assert np.all(backend.is_vacuum(tol))
 
-    def test_mean_and_std_vacuum(self, setup_backend, tol):
+    def test_mean_and_std_vacuum(self, setup_backend, pure, tol):
         """Tests that the mean and standard deviation estimates of many homodyne
         measurements are in agreement with the expected values for the
         vacuum state"""
-
         x = np.empty(0)
+
+        backend = setup_backend(1)
+
         for _ in range(N_MEAS):
-            backend = setup_backend(1)
+            backend.reset(pure=pure)
```

Comment:
Fixed this test so that it doesn't leak memory.

* Moved `setup_backend` outside the loop
* reset backend at beginning of every loop

This pattern is repeated every time this issue appears.